### PR TITLE
ENH add ksigma moments

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds the new ksigma moments for ngmix.

### CODE REVIEW

**Comment by beckermr** [comment]
@esheldon This PR has the new ksigma moments. I adjusted the code so that they will only work with ngmix v2 when it is installed and will fail otherwise. I adjusted the tests to respect that and separated the case.

File: metadetect/fitting.py [hunk 1]
```diff
@@ -346,6 +346,92 @@ def _check_flags(self, mbobs):
...
+
+        if obs.has_psf():
+            res = self._fitter.go(obs)
+        else:
+            res = self._fitter.go(obs, no_psf=True)
+
+        res['numiter'] = 1
+        res['g'] = res['e']
+        res['g_cov'] = res['e_cov']
+        res['g_err'] = res['e_err']
```
Comment (esheldon) [inline metadetect/fitting.py:375]: are you copying this out?  Should we add that to _measure_moments in the Moments class too?

File: metadetect/fitting.py [hunk 2]
```diff
@@ -346,6 +346,92 @@ def _check_flags(self, mbobs):
...
+            output[n('raw_mom')] = res['mom']
+            output[n('raw_mom_cov')][:] = res['mom_cov']
+        else:
+            flags |= procflags.NOMOMENTS_FAILURE
+
+        if res['flags'] == 0:
+            output[n('s2n')] = res['s2n']
+            output[n('pars')] = res['pars']
+            output[n('g')] = res['g']
+            output[n('g_cov')] = res['g_cov']
```
Comment (esheldon) [inline metadetect/fitting.py:416]: I don't see g_err copied out here

**Review by esheldon — APPROVED** [review]
looks good, minor comment

File: metadetect/fitting.py (hunk 1 above)
Comment (beckermr) [reply metadetect/fitting.py]: I removed this.

**Pull request merged**